### PR TITLE
Set seed value using CLI

### PR DESCRIPTION
**Pull request by adityapande-1995**
🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

# 🎉 New feature

Closes https://github.com/gazebosim/gz-sim/issues/1467

## Summary
This PR exposes a ``--seed`` argument to ``gz sim`` to set seed value for the random number generator.

## Test it
Added a small test case that runs ``gz sim -s -v 4 --seed 5 --iterations 5`` and checks for the output string "Setting seed value: 5" in stdout.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: src/cmd/cmdsim.rb.in [hunk 1]
```diff
@@ -106,6 +106,9 @@ COMMANDS = { 'sim' =>
   "  --log-compress               When recording, compress final log files.        \n"\
   "                               Only valid if recording is enabled.              \n"\
   "\n"\
+  "  --seed                       Pass a custom seed value to the random           \n"\
```
Comment (chapulina) [inline src/cmd/cmdsim.rb.in:109]:
```suggestion
  "  --seed  [arg]                Pass a custom seed value to the random           \n"\
```

**Review by chapulina — COMMENTED** [review]
LGTM so far, just pending a test

File: src/cmd/cmdsim.rb.in (hunk 1 above)
Comment (adityapande-1995) [reply src/cmd/cmdsim.rb.in]: Done https://github.com/gazebosim/gz-sim/pull/1618/commits/c918e9b9128c62e41992a2b1bb3bdccc2802e081

**Comment by chapulina** [comment]
Removed the `Garden` label because we're past feature freeze ❄️ .

This PR can either be merged into `main` and backported to `gz-sim7` after the stable release, or retargeted to `gz-sim7` and merged after the stable release (i.e. October).

**Review by nkoenig — APPROVED** [review]

**Comment by adityapande-1995** [comment]
The CI failures here seem unrelated to the changes made here. Also, this PR breaks ABI, so we can't backport this to Garden (gz-sim7).

**Pull request merged**